### PR TITLE
Bug 1777255 - Make mobile version numbers use the same pattern as Gecko

### PR DESCRIPTION
**Pull request by JohanLorenzo**

### CODE REVIEW

File: mozilla_version/mobile.py [hunk 1]
```diff
@@ -42,30 +46,97 @@ class MobileVersion(BaseVersion):
...
+
+        if self.is_gecko_pattern:
+            error_messages.extend([
+                pattern_message
+                for condition, pattern_message in ((
+                    self.beta_number is not None and self.patch_number is not None,
+                    'Beta number and patch number cannot be both defined',
+                ), (
+                    self.release_candidate_number is not None,
+                    'Release candidate number cannot be defined starting Mobile v104',
```
Comment (gbrownmozilla) [inline mozilla_version/mobile.py:85]: nit: Error message seems a little awkward. Maybe "...starting with Mobile v104" or "...after..."?

File: mozilla_version/mobile.py [hunk 2]
```diff
@@ -42,30 +46,97 @@ class MobileVersion(BaseVersion):
...
+        mobile_version = super(MobileVersion, cls).parse(
+            version_string, regex_groups=('is_nightly',)
+        )
+
+        # Betas are supported in both the old and the gecko pattern. Let's make sure
+        # the string we got follows the right rules
+        if mobile_version.is_beta:
+            if mobile_version.is_gecko_pattern and '-beta.' in version_string:
+                raise PatternNotMatchedError(
+                    mobile_version, ['"-beta." cannot be used starting Mobile v104']
```
Comment (gbrownmozilla) [inline mozilla_version/mobile.py:123]: nit: awkward message

**Review by gbrownmozilla — APPROVED** [review]
lgtm. I like the test consolidation, and can't find anything wrong here.

File: mozilla_version/mobile.py (hunk 1 above)
Comment (JohanLorenzo) [reply mozilla_version/mobile.py:85]: Thank you very much! I do enjoy this kind of feedback. It helps me speak/write better English every day 👍 Last year, I actually read this eBook[1] and I was like "ohhhh, I've made this mistake for so many years!". 

So, keep this feedback coming! 😃 

[1] https://jakubmarian.com/english-mistakes/

File: mozilla_version/mobile.py (hunk 2 above)
Comment (JohanLorenzo) [reply mozilla_version/mobile.py:123]: Changed into `"-beta." can only be used before Mobile v104`. Let me know if you'd like me to rephrase it. I'm happy to do it in a follow-up.

**Pull request merged**